### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by Mrtenz**
This improves error handling by:

- Using `@metamask/rpc-errors` instead of `eth-rpc-errors` throughout the entire stack.
- Introducing a new `SnapError`, which will not cause the Snap to crash when thrown.
- Unwrapping errors where possible.
- Also bumps a ton of MetaMask dependencies, see https://github.com/MetaMask/snaps/pull/1818

### CODE REVIEW

File: packages/snaps-controllers/package.json
Comment (FrederikBolding) [inline packages/snaps-controllers/package.json:141]: I think these can be removed. AFAIK we only need to specify these at the root package.json

**Comment by Mrtenz** [comment]
@SocketSecurity ignore eth-block-tracker@8.1.0

This is ours.

File: packages/snaps-controllers/package.json
Comment (FrederikBolding) [inline packages/snaps-controllers/package.json:52]:
```suggestion
```
It's a peer dep now

File: packages/snaps-controllers/src/services/AbstractExecutionService.ts
Comment (hmalik88) [inline packages/snaps-controllers/src/services/AbstractExecutionService.ts:223]: Is the cast not necessary any more? If so, above comment should be removed.

File: packages/snaps-controllers/src/services/AbstractExecutionService.ts
Comment (hmalik88) [inline packages/snaps-controllers/src/services/AbstractExecutionService.ts:403]: why is casting needed here?

Comment (Mrtenz) [reply packages/snaps-controllers/src/services/AbstractExecutionService.ts:403]: `request` is typed as `Record<string, unknown>`, but AFAIK we validate that it's a `JsonRpcRequest` before reaching this point.

File: packages/snaps-controllers/src/services/node/NodeProcessExecutionService.test.ts
Comment (hmalik88) [inline packages/snaps-controllers/src/services/node/NodeProcessExecutionService.test.ts:78]: Why are these lines needed?

Comment (Mrtenz) [reply packages/snaps-controllers/src/services/node/NodeProcessExecutionService.test.ts:78]: TypeScript can't infer the type of errors (since anything is technically a valid error). Jest's `.toBeInstanceOf` also doesn't do any type-level assertions. I'm not sure why we need `.toEqual` here, but the tests fail with `.toStrictEqual`.

File: packages/snaps-controllers/src/services/AbstractExecutionService.ts
Comment (FrederikBolding) [reply packages/snaps-controllers/src/services/AbstractExecutionService.ts:403]: Yeah, we do, we just don't have a better type right here

**Review by FrederikBolding — COMMENTED** [review]
LGTM

Holding off on approving until someone else has approved.

File: packages/snaps-execution-environments/src/common/endowments/endowments.test.browser.ts
Comment (hmalik88) [inline packages/snaps-execution-environments/src/common/endowments/endowments.test.browser.ts:28]: what's this change for?

Comment (FrederikBolding) [reply packages/snaps-execution-environments/src/common/endowments/endowments.test.browser.ts:28]: To mirror the LM/SES config we run the execution environments under: https://github.com/MetaMask/snaps/blob/main/.yarn/patches/%40lavamoat-lavapack-npm-5.4.1-2548b3e225.patch

I believe Maarten was getting an error in tests but not when actually using the execution environments because of this mismatch

Comment (Mrtenz) [reply packages/snaps-execution-environments/src/common/endowments/endowments.test.browser.ts:28]: I think we could revert this now, since we removed the `SnapError` endowment later on, but regardless it's good for the config we use here to match LavaMoat.

**Review by hmalik88 — APPROVED** [review]
LGTM. Great work, guys 🚀

**Pull request merged**
